From ae16e2d56191d1d23f4b79b9d9c01f4aff0bc847 Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@st.com>
Date: Tue, 21 May 2024 20:48:41 +0200
Subject: [PATCH 599/841] fix(st-clock): display proper PLL number

The PLL clk_id does not start at 0, but it is in the enum listing all
clocks. To have a better display of the PLL number, start at PLL1,
by changing pll->clk_id in messages with pll->clk_id - _CK_PLL1 + 1.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git a0097aeb03491812d59c4648168f1822e7a89aef fix(st-clock): display proper PLL number for STM32MP13
2d79c8e6f093f775fbd7afbe2e9221e2f343c7a0 fix(st-clock): display proper PLL number]
Signed-off-by: Yann Gautier <yann.gautier@st.com>
Change-Id: Ie5b123a4197466df40a86ea8cd6880669912b160
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/381501
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Reviewed-by: Gabriel FERNANDEZ <gabriel.fernandez@foss.st.com>
Tested-by: Yann GAUTIER <yann.gautier@foss.st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 drivers/st/clk/clk-stm32mp2.c | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/drivers/st/clk/clk-stm32mp2.c b/drivers/st/clk/clk-stm32mp2.c
index 87ab4cc97..4f3fdef5d 100644
--- a/drivers/st/clk/clk-stm32mp2.c
+++ b/drivers/st/clk/clk-stm32mp2.c
@@ -802,8 +802,9 @@ static int _clk_stm32_pll_wait_ready_on(struct stm32_clk_priv *priv,
 	/* Wait PLL lock */
 	while ((mmio_read_32(pllxcfgr1) & RCC_PLLxCFGR1_PLLRDY) == 0U) {
 		if (timeout_elapsed(timeout)) {
-			ERROR("CLK %d : PLL start failed @ 0x%x: 0x%x\n",
-			      pll->clk_id, pll->reg_pllxcfgr1, mmio_read_32(pllxcfgr1));
+			ERROR("PLL%d start failed @ 0x%x: 0x%x\n",
+			      pll->clk_id - _CK_PLL1 + 1, pll->reg_pllxcfgr1,
+			      mmio_read_32(pllxcfgr1));
 			return -ETIMEDOUT;
 		}
 	}
@@ -821,7 +822,7 @@ static int _clk_stm32_pll_wait_ready_off(struct stm32_clk_priv *priv,
 	while ((mmio_read_32(pllxcfgr1) & RCC_PLLxCFGR1_PLLRDY) != 0U) {
 		if (timeout_elapsed(timeout)) {
 			ERROR("PLL%d stop failed @ 0x%lx: 0x%x\n",
-			      pll->clk_id, pllxcfgr1, mmio_read_32(pllxcfgr1));
+			      pll->clk_id - _CK_PLL1 + 1, pllxcfgr1, mmio_read_32(pllxcfgr1));
 			return -ETIMEDOUT;
 		}
 	}
@@ -1785,7 +1786,7 @@ static int clk_stm32_pll_wait_mux_ready(struct stm32_clk_priv *priv,
 	while ((mmio_read_32(pllxcfgr1) & RCC_PLLxCFGR1_CKREFST) !=
 	       RCC_PLLxCFGR1_CKREFST) {
 		if (timeout_elapsed(timeout)) {
-			EARLY_ERROR("PLL%d ref clock not started\n", pll->clk_id);
+			EARLY_ERROR("PLL%d ref clock not started\n", pll->clk_id - _CK_PLL1 + 1);
 			return -ETIMEDOUT;
 		}
 	}
-- 
2.47.2

